### DISCUSSION TITLE
File Hashes?

### DISCUSSION THREAD

**Discussion by LegendaryB** [pnp/pnpcore#600] [Q&A] [answered]
Does PnP.Core contains this functionality or is it only available through manual API call to MS graph?
https://docs.microsoft.com/en-us/onedrive/developer/rest-api/resources/hashes?view=odsp-graph-online

**Comment by jansenbe** [comment, marked as answer]
@LegendaryB , only via making the Graph call yourselves at the moment. Graph call to make uses this structure:

`https://graph.microsoft.com/v1.0/drives/<driveid>/items/<driveitemid>/file`. You can get the `driveid` and `driveitemid` values from the IFile object via properties `VroomDriveId` and `VroomItemId`.

To verify if a local file version matches with the cloud one you need to also implement the QuickXOR hash algorithm as explained here https://docs.microsoft.com/en-us/onedrive/developer/code-snippets/quickxorhash?view=odsp-graph-online.

**Reply by LegendaryB** [reply]
Awesome thank you! 👍

**Reply by DaleyKD** [reply]
Gotta say, I sure wish SharePoint REST had this. All that work we've been doing with FileVerisons... having the hash could've easily saved us some processing cycles. 😄
